backend/compact-connect/lambdas/purchases/purchase_client.py

```diff
@@ -410,3 +463,56 @@ def void_privilege_purchase_transaction(self, compact_name: str, order_informati
             )
 
         return self.payment_processor_client.void_unsettled_charge_on_credit_card(order_information=order_information)
+
+    def validate_and_store_credentials(self, compact_name: str, credentials: dict) -> dict:
+        """
+        Validate the provided payment credentials and store them in secrets manager.
+
+        :param compact_name: The name of the compact
+        :param credentials: The payment processor credentials
+        :return: A response indicating the credentials were validated and stored successfully
+        :raises CCInvalidRequestException: If the credentials are invalid
+        """
+        if credentials['processor'] != AUTHORIZE_DOT_NET_CLIENT_TYPE:
+            raise CCInvalidRequestException('Invalid payment processor')
+
+        # call payment processor test endpoint to validate the credentials
+        # if the credentials are invalid, authorize.net will return an error response
+        secret_value = {
+            'processor': credentials['processor'],
+            'api_login_id': credentials['apiLoginId'],
+            'transaction_key': credentials['transactionKey'],
+        }
+        # this will raise an exception if the credentials are invalid
+        response = PaymentProcessorClientFactory().create_payment_processor_client(secret_value).validate_credentials()
+
+        # no exceptions were raised, so the credentials are valid
+        # first check to see if secret already exists
+        try:
+            self.secrets_manager_client.describe_secret(
+                SecretId=self._get_payment_processor_secret_name_for_compact(compact_name))
+
+            # secret exists, update its value to whatever the admin sent us
+            logger.info('Existing secret found, updating secret for compact', compact_name=compact_name)
+            self.secrets_manager_client.put_secret_value(
+                SecretId=self._get_payment_processor_secret_name_for_compact(compact_name),
+                SecretString=json.dumps(secret_value)
+            )
+        except self.secrets_manager_client.exceptions.ResourceNotFoundException:
+            # secret does not exist, so we can create it
+            logger.info('Existing secret not found, creating new secret for compact', compact_name=compact_name)
+            self.secrets_manager_client.create_secret(
+                Name=self._get_payment_processor_secret_name_for_compact(compact_name),
+                SecretString=json.dumps(secret_value)
+            )
+
+        return response
+
+
+if __name__ == '__main__':
+    purchase_client = PurchaseClient()
+    purchase_client.validate_and_store_credentials("count", {
+        "processor": "authorize.net",
+        "apiLoginId": "24xMHMr4L5TX",
+        "transactionKey": "2e732TCu3hm8Q2FF"
```

Comment:
should this be in the code?

Reply:
Nope, rookie mistake from local testing 🤦 Look like I'll need to go roll the sandbox credentials now. 😆